CONTRIBUTING_WITH_GIT2R.Rmd

```diff
@@ -0,0 +1,551 @@
+between a PAT and a password is that a PAT can be stored as text on your
+computer (which you should NEVER do with passwords) because it is easy to
+generate and (most importantly) easy to remove. We can create a PAT from GitHub
+and then we will store this in a file called `.Renviron` that is read every time
+you restart your R session, allowing you to have this token whenever you need to
+push changes to your fork on GitHub. [The instructions to generate your PAT are
+here](https://help.github.com/articles/creating-an-access-token-for-command-line-use/).
+
+Here's an example of what a `.Renviron` file with a PAT will look like on the
+inside:
+
+```
+GITHUB_PAT="97cc6bf86c31a42fca2de32884cd1f1c4b1102ba"
+
+
+
+
+```
+
+Once you have created your PAT, make a text file called `.Renviron` inside of
+the repository and paste your PAT in that file as shown above. This will not be
+tracked by git, and will only exist on your computer.
+
+```{r renviron_shadow, echo = FALSE}
+cat(paste0("GITHUB_PAT=\"¯\\_(ツ)_/¯\""), file = paste0(fork_pgi, "/.Renviron"))
+```
+
+The next time you open R from `popgenInfo.Rproj`, this token will be available
+for GitHub to ensure that you are who you say you are. It is important to note
+that, if you change or delete that PAT, you must replace it here.
+
+In the next step, we will ensure that our fork is up to date with NESCent. In 
+order to do that, we need to tell our local copy that it can take updates from 
+NESCent. We do this by adding NESCent as a **remote** using the function 
+`remote_add()` and checking to make sure it worked with the function
+`remotes()`.
+
+```{r remote_add}
+remote_add(repo, 
+           name = "NESCent", 
+           url = "https://github.com/NESCent/popgenInfo.git")
+
+# Check our remotes
+remotes(repo)
+```
+
+With all of this information set, we can begin working on our repository.
+
+# Step 3: Keeping your fork up to date
+
+The first thing we need to do is make sure that our master branch is up to date.
+Since other people are also contributing to this project, it is possible for
+changes to be made even within seconds of you forking the repository. In this
+section we will show you how you can keep your master branch up to date in your
+local repository and your fork on github. 
+
+Let's assume that you've just opened a fresh Rstudio session with the
+`popgenInfo.Rproj` file. First, let's load *git2r* and access our repository.
+
+```{r git2r_show, eval = FALSE}
+library("git2r")
+repo <- repository()
+```
+
+Now that we have our repository loaded, we should make sure that we are on the
+**master branch**. We can do this using the `checkout()` function from *git2r*.
+
+```{r checkout_master}
+checkout(repo, "master")
+repo
+```
+
+Now, whether or not we suspect changes from NESCent, it's always good form to 
+update your local repository before creating any new branches. We can do it this
+by first **fetch**ing the NESCent master branch and then **merging** it into our
```

Comment:
Going off of that point: what's your opinion on keeping this as a dynamic Rmd file or a static md file? the NESCent repo and my fork will be in various states each time the website is rebuilt, so what appears to the user will always be different if it's kept as a dynamic Rmd file.